src/keygen/participant.rs

```diff
@@ -462,37 +462,37 @@ impl KeygenParticipant {
         proof.verify_with_transcript(&input, &transcript)?;
         let keyshare = decom.get_keyshare();
         self.storage.store(
-            StorableType::PublicKeyshare,
+            MainStorageType::PublicKeyshare,
             message.id(),
             message.from(),
             &keyshare,
         )?;
 
         //check if we've stored all the public keyshares
         let keyshare_done = self.storage.contains_for_all_ids(
-            StorableType::PublicKeyshare,
+            MainStorageType::PublicKeyshare,
             message.id(),
             &[self.other_participant_ids.clone(), vec![self.id]].concat(),
         )?;
 
         if keyshare_done {
             for oid in self.other_participant_ids.iter() {
-                self.storage.transfer::<StorableType, KeySharePublic>(
+                self.storage.transfer::<MainStorageType, KeySharePublic>(
```

Comment:
Rust can infer the first type argument to the transfer functions here. To avoid repetition, I think we should just use a <_, KeySharePublic> here.